packages/steiger-plugin-fsd/src/no-wildcard-exports/index.spec.ts

```diff
@@ -0,0 +1,258 @@
+          fileWithContent.content = "export * as userModel from './model'"
+        } else if (child.path === joinFromRoot('features', 'auth', 'ui', 'index.ts')) {
+          fileWithContent.content = "export { LoginForm } from './LoginForm'"
+        } else {
+          fileWithContent.content = ''
+        }
+      } else if (child.type === 'folder') {
+        addContentToFiles(child)
+      }
+    }
+  }
+
+  addContentToFiles(root)
+
+  expect(noWildcardExports.check(root)).toEqual({ diagnostics: [] })
+})
+
+it('reports errors on a project with wildcard exports', () => {
+  const root = parseIntoFsdRoot(`
+    📂 shared
+      📂 ui
+        📄 Button.tsx
+        📄 index.ts
+    📂 entities
+      📂 user
+        📂 ui
+          📄 UserCard.tsx
+          📄 index.ts
+        📄 index.ts
+  `)
+
+  function addContentToFiles(folder: Folder): void {
+    for (const child of folder.children) {
+      if (child.type === 'file') {
+        const fileWithContent = child as FileWithContent
+        if (child.path === joinFromRoot('shared', 'ui', 'index.ts')) {
+          fileWithContent.content = "export * from './Button'"
+        } else if (child.path === joinFromRoot('entities', 'user', 'ui', 'index.ts')) {
+          fileWithContent.content = "export * from './UserCard'"
+        } else {
+          fileWithContent.content = ''
+        }
+      } else if (child.type === 'folder') {
+        addContentToFiles(child)
+      }
+    }
+  }
+
+  addContentToFiles(root)
+
+  const diagnostics = noWildcardExports.check(root).diagnostics
+  expect(diagnostics).toEqual([
+    {
+      message: 'Wildcard exports are not allowed in public APIs. Use named exports instead.',
+      location: { path: joinFromRoot('entities', 'user', 'ui', 'index.ts') },
+      fixes: [
+        {
+          type: 'modify-file',
+          path: joinFromRoot('entities', 'user', 'ui', 'index.ts'),
+          content: '// Replace with named exports\n// Example: export { ComponentA, ComponentB } from "./components"',
+        },
+      ],
+    },
+  ])
+})
+
+it('allows export * as namespace pattern', () => {
+  const root = parseIntoFsdRoot(`
+    📂 shared
+      📂 ui
+        📄 positions.ts
+        📄 index.ts
+    📂 entities
+      📂 user
+        📄 model.ts
+        📄 index.ts
+  `)
+
+  function addContentToFiles(folder: Folder): void {
+    for (const child of folder.children) {
+      if (child.type === 'file') {
+        const fileWithContent = child as FileWithContent
+        if (child.path === joinFromRoot('shared', 'ui', 'index.ts')) {
+          fileWithContent.content = "export * as positions from './positions'"
+        } else if (child.path === joinFromRoot('entities', 'user', 'index.ts')) {
+          fileWithContent.content = "export * as userModel from './model'"
+        } else {
+          fileWithContent.content = ''
+        }
+      } else if (child.type === 'folder') {
+        addContentToFiles(child)
+      }
+    }
+  }
```

Comment:
issue: I think this conceals the fact that this test will actually not pass during real usage, since the linter engine doesn't include the `content` field

suggestion: let's use FS mocking for tests, like it's done in the [test file for `forbidden-imports`](https://github.com/feature-sliced/steiger/blob/1281d600506ae5f58479ca1266c53201c3a44247/packages/steiger-plugin-fsd/src/forbidden-imports/index.spec.ts)